### PR TITLE
Address the Multi-Speed DX heating coil issue with HSPF2 calculation

### LINKED ISSUE #10587: Multi-Speed heating coil issue with HSPF 2
Issue overview
--------------
I tested a defect file of a helpdesk ticket 16700 in EnergyPlus Support and confirmed there is a problem how the HSPF is calculated. The HSPF and HSPF2 shouldn't be significantly different. This is a defect. 

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (develop, [e2c4305](https://github.com/NREL/EnergyPlus/commit/e2c430549a6c69dd555668fa10b0fdc18883a873))
 - helpdesk ticket number 16700

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to EnergyPlus Defect Complexity (Github Project)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - Fixes #10587
 - Corrected HSPF and HSPF2 calculations due to incorrect function argument and failure to set a value to a local variable
 - HSPF and HSPF2 are now reported separately in EIO and HTML tabular output files
 - Added units to DX Heating Coils equipment summary table `Minimum Outdoor Dry-Bulb Temperature for Compressor Operation` column header. 

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by Nigusse** [comment]
The investigation of helpdesk ticket led to two issues:

(1) The arguments `OATempCompressorOn` and `OATempCompressorOnOffBlank` of the `MultiSpeedDXHeatingCoilHSPF2` and `MultiSpeedDXHeatingCoilHSPF` functions were swapped. As the results of the swap incorrect values were being passed.

(2) After fixing swamp the HSPF and HSPF2 were further apart hence needed further investigation, which led to uncover that the local variable `OATempCompressorOff` was not set a value in the `MultiSpeedDXHeatingCoilHSPF2` function. The local variable rather uses initialized value of `0` all the time.

These changes impact HSPF values in multi-speed DX heating coils.

**Comment by Nigusse** [comment]
HSPF values before the fix (Develop):
image

image

Results after correcting the function argument swamp (Issue 1):

image

image

Results after correcting the arguments swap (Issue 1) and correctly assigning a value to the local variable ` OATempCompressorOff = MinOATCompressor;` (Issue 2):

image

image


After addressing issues 1 and 2, the HSPF and HSPF2 values are match better aligned. Now the HSPF2 value is about 18% lower compared to about 29% lower before the fix.

**Comment by Nigusse** [comment]
@mjwitte This PR uncovered and addressed a defect and made the HSPF2 value more reasonable, but I am not 100% convinced that the 18% difference between HSPF and HSPF2 in this defect file is acceptable. Let's wait and see how much the HSPF2 values diffs are for the regression test suite.

**Comment by Nigusse** [comment]
Intermediate results of HSPF2 calculation for develop: 
The `0` resistance heater (rh) value at -2.78C OA temperature triggered further investigation that uncovered issue 1. 

image


Intermediate results with issue 1 addressed:
image

Now the rh values are in a consistent trend but the SHPF2 value even gotten lower, which triggered further investigation and led to uncover the local variable not set a value issue 2.

Intermediate results with issue 2 addressed:
image

**Comment by Nigusse** [comment]
Modified a unit test HSPF2 value such that a failed unit test passes. There are 19 test suite IDFs that show regression diffs in the tabular  and/or eio output files due to HSPF and/or HSPF2 values change. I will examine selected test suite regression results diffs.

**Comment by Nigusse** [comment]
The regression results diffs for the example files `AirflowNetwork_MultiZone_House_TwoSpeed.idf` and `EMSMultiSpeedHeatPumpOverrideControl.idf` are the same and are shown below.

```
 ! , Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF {Btu/W-h}, Region Number
- DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 13652.1, 8840.9, 8.91, 4
- DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 13849.2, 9038.1, 6.45, 4
+ DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 13652.1, 8840.9, 8.28, 4
+ DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 13849.2, 9038.1, 7.00, 4
```

HSPF2 value increased from `6.45` to `7.0`. However, the HSPF value decreased from `8.91` to `8.28`. Note that the HSPF value is affected by issue 1 only.

**Comment by Nigusse** [comment]
In the example file `MultiSpeedHeatPump.idf`, both SHPF and SHPF2 values decreased, and the difference was slightly narrowed.

```
 ! , Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF {Btu/W-h}, Region Number
- DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 36720.8, 20238.3, 6.61, 4
- DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 37178.2, 20695.8, 5.23, 4
+ DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 36720.8, 20238.3, 5.91, 4
+ DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 37178.2, 20695.8, 5.13, 4
```

**Comment by Nigusse** [comment]
Selected regression result diffs were examined to show that the difference between HSPF and HSPF2 has narrowed regardless of their values' change trend.

**Comment by Nigusse** [comment]
The 19 test suite files showing the difference are those with multi-speed DX heating coils. Diffs were observed in `tabular` and `eio` output files.

File: src/EnergyPlus/StandardRatings.cc [hunk 1]
```diff
@@ -6266,6 +6266,9 @@ namespace StandardRatings {
         Real64 e_sum(0.0);
         Real64 rh_sum(0.0);
 
+        // The minimum temperature below which the compressor is turned off
+        OATempCompressorOff = MinOATCompressor;
```
Comment (Nigusse) [inline src/EnergyPlus/StandardRatings.cc:6270]: The local variable `OATempCompressorOff` was not set a value and the initialized value of `0` was used.

File: src/EnergyPlus/StandardRatings.cc [hunk 2]
```diff
@@ -6646,8 +6649,8 @@ namespace StandardRatings {
                                                                                                             RatedCOP,
                                                                                                             RegionNum,
                                                                                                             MinOATCompressor,
-                                                                                                            OATempCompressorOnOffBlank,
                                                                                                             OATempCompressorOn,
+                                                                                                            OATempCompressorOnOffBlank,
```
Comment (Nigusse) [inline src/EnergyPlus/StandardRatings.cc:6653]: The function arguments needed to swap their order to match the argument's order in the function declaration.

File: src/EnergyPlus/StandardRatings.cc [hunk 3]
```diff
@@ -6668,8 +6671,8 @@ namespace StandardRatings {
                                          RatedCOP,
                                          RegionNum,
                                          MinOATCompressor,
-                                         OATempCompressorOnOffBlank,
                                          OATempCompressorOn,
+                                         OATempCompressorOnOffBlank,
                                          DefrostControl);
```
Comment (Nigusse) [inline src/EnergyPlus/StandardRatings.cc:6676]: The function arguments needed to swap their order to match the argument's order in the function declaration.

**Review by Nigusse — COMMENTED** [review]
Multispeed DX Heating coil HSPF value calculation code changes walkthrough.

**Comment by mjwitte** [comment]
@Nigusse Thanks for sorting this out. There's one other thing that needs a fix. In the eio output, there are two rows with the same label and header row. One writes HSPF, and the second one writes the HSPF2 value in the HSPF column. From example file MultiSpeedHeatPump:
```
! <DX Heating Coil Standard Rating Information>, Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF {Btu/W-h}, Region Number
 DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 36720.8, 20238.3, 5.91, 4
 DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 37178.2, 20695.8, 5.13, 4
```
As echoed in the table output:
image

For the cooling ratings, the header and line label are different so you can tell what's what.
```
! <DX Cooling Coil Standard Rating Information>, Component Type, Component Name, Standard Rating (Net) Cooling Capacity {W}, Standard Rated Net COP {W/W}, EER {Btu/W-h}, SEER User {Btu/W-h}, SEER Standard {Btu/W-h}, IEER {Btu/W-h}
 DX Cooling Coil Standard Rating Information, Coil:Cooling:DX:MultiSpeed, HEAT PUMP ACDXCOIL 1, 34330.8, 2.71, 9.24, 10.72, 10.67, 0
! <DX Cooling Coil AHRI 2023 Standard Rating Information>, Component Type, Component Name, Standard Rating (Net) Cooling Capacity {W}, Standard Rating Net COP2 {W/W}, EER2 {Btu/W-h}, SEER2 User {Btu/W-h}, SEER2 Standard {Btu/W-h}, IEER 2022 {Btu/W-h}
 DX Cooling Coil AHRI 2023 Standard Rating Information, Coil:Cooling:DX:MultiSpeed, HEAT PUMP ACDXCOIL 1, 29101.0, 2.42, 8.25, 9.56, 9.90, 9.7
```

1. The heating ratings should have a new header line and write a different line label, e.g.:
```
! <DX Heating Coil Standard Rating Information>, Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF {Btu/W-h}, Region Number
 DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 36720.8, 20238.3, 5.91, 4
! <DX Heating Coil AHRI 2023 Standard Rating Information>, Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF2 {Btu/W-h}, Region Number
 DX Heating Coil AHRI 2023 Standard Rating Information, Coil:Heating:DX:MultiSpeed, HEAT PUMP DX HEATING COIL 1, 37178.2, 20695.8, 5.13, 4
```

2. In the Equipment Summary tables, these are better, but maybe the subtable title should change from "DX Heating Coils [ HSPF2 ]" to something like "DX Heating Coil Standard Ratings 2023"?
image

3. Is it correct that the rated capacity is different in the 2023 ratings (both for cooling and for heating)?

**Comment by Nigusse** [comment]
@mjwitte Ok. I will work on items 1 and 2. Regarding item 3, my understanding is that they can be different but depends on the user specified values for the following fields: 

```
    636,                     !- 2017 Speed 1 Rated Supply Air Fan Power Per Volume Flow Rate {W/(m3/s)}
    636,                     !- 2023 Speed 1 Rated Supply Air Fan Power Per Volume Flow Rate {W/(m3/s)} 
```
If these values are the same, then net rated capacities must be the same and vice versa.

**Comment by Nigusse** [comment]
Addressed issue 1: now separate line of header and data streams for the respective AHRI versions in the `eio` output files.

```
! <DX Heating Coil Standard Rating Information>, Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF {Btu/W-h}, Region Number
 DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, 9770.6, 5770.5, 7.99, 4
! <DX Heating Coil AHRI 2023 Standard Rating Information>, Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF2 {Btu/W-h}, Region Number
 DX Heating Coil AHRI 2023 Standard Rating Information, Coil:Heating:DX:MultiSpeed, DX HEATING COIL_UNIT1, 9770.6, 5770.5, 6.52, 4
```

**Comment by Nigusse** [comment]
Addressed issue 1: separate tables for the DX Heating Coils AHRI standard ratings versions in tabular output file.

image

**Comment by Nigusse** [comment]
Address issue 2: Changed Equipment Summary Table Header of DX Heating Coils for 2023 AHRI Standard Ratings

image

I will push an updated branch that reflects the above table. The current CI run does not have the last two columns (Minimum OA Temperature and Airloop Name). Note that the table named slightly differently from what you suggested. I will change it if you don't agree, just let me know @mjwitte.

**Comment by rraustad** [comment]
I noticed that the Minimum Outdoor Dry-Bulb Temperature does not have units [C or F].

**Comment by Nigusse** [comment]
With the units for the `Minimum OA Dry-Bulb Temperature ...`:

image

**Comment by Nigusse** [comment]
This branch is ready for review. Below is summary of the expected diffs:
(1) HSPF and HSPF2 value diffs because of the fix. 19 test suite files have diffs in `html` and `eio` output files. 
(2) Tabular structural diffs due to new and modified tables. Separate HSPF and HSPF2 tables in tabular output files.
(3) Renamed an existing table and modified column heading by adding units to DX Heating Coils tables
(4) Audits warning diffs due to column count change

File: src/EnergyPlus/DXCoils.cc [hunk 4]
```diff
@@ -8691,6 +8691,14 @@ void SizeDXCoil(EnergyPlusData &state, int const DXCoilNum)
...
                                                  thisDXCoil.AirLoopNum > 0 ? state.dataAirSystemsData->PrimaryAirSystems(thisDXCoil.AirLoopNum).Name
                                                                            : "N/A");
+        // std 229 existing table DX Heating coil 2023 AHRI new reporting variables
+        OutputReportPredefined::PreDefTableEntry(
+            state, state.dataOutRptPredefined->pdchDXHeatCoilMinOADBTforCompOp_2023, equipName, thisDXCoil.MinOATCompressor);
+        OutputReportPredefined::PreDefTableEntry(state,
+                                                 state.dataOutRptPredefined->pdchDXHeatCoilAirloopName_2023,
+                                                 equipName,
+                                                 thisDXCoil.AirLoopNum > 0 ? state.dataAirSystemsData->PrimaryAirSystems(thisDXCoil.AirLoopNum).Name
+                                                                           : "N/A");
```
Comment (Nigusse) [inline src/EnergyPlus/DXCoils.cc:8701]: This code change adds columns for the new equipment summary table of the DX Heating Coils 2023 AHRI.

File: src/EnergyPlus/OutputReportPredefined.cc [hunk 5]
```diff
@@ -388,16 +388,20 @@ namespace OutputReportPredefined {
         s->pdchDXHeatCoilRegionNum = newPreDefColumn(state, s->pdstDXHeatCoil, "Region Number");
         // Std 229 Predef outputs for DX Heating Coils
         s->pdchDXHeatCoilMinOADBTforCompOp =
-            newPreDefColumn(state, s->pdstDXHeatCoil, "Minimum Outdoor Dry-Bulb Temperature for Compressor Operation");
+            newPreDefColumn(state, s->pdstDXHeatCoil, "Minimum Outdoor Dry-Bulb Temperature for Compressor Operation [C]");
         s->pdchDXHeatCoilAirloopName = newPreDefColumn(state, s->pdstDXHeatCoil, "Airloop Name");
```
Comment (Nigusse) [inline src/EnergyPlus/OutputReportPredefined.cc:409]: Adds units to a column heading.

File: src/EnergyPlus/OutputReportPredefined.cc [hunk 6]
```diff
@@ -388,16 +388,20 @@ namespace OutputReportPredefined {
...
 
         // for DX Heating Coil AHRI Standard 2023 Ratings | HSPF2
-        s->pdstDXHeatCoil_2023 = newPreDefSubTable(state, s->pdrEquip, "DX Heating Coils [ HSPF2 ]");
+        s->pdstDXHeatCoil_2023 = newPreDefSubTable(state, s->pdrEquip, "DX Heating Coils AHRI 2023");
```
Comment (Nigusse) [inline src/EnergyPlus/OutputReportPredefined.cc:412]: Renamed an existing table name.

File: src/EnergyPlus/OutputReportPredefined.cc [hunk 7]
```diff
@@ -388,16 +388,20 @@ namespace OutputReportPredefined {
...
         s->pdchDXHeatCoilType_2023 = newPreDefColumn(state, s->pdstDXHeatCoil_2023, "DX Heating Coil Type");
         s->pdchDXHeatCoilHighCap_2023 = newPreDefColumn(state, s->pdstDXHeatCoil_2023, "High Temperature Heating (net) Rating Capacity [W]");
         s->pdchDXHeatCoilLowCap_2023 = newPreDefColumn(state, s->pdstDXHeatCoil_2023, "Low Temperature Heating (net) Rating Capacity [W]");
         s->pdchDXHeatCoilHSPF2IP_2023 = newPreDefColumn(state, s->pdstDXHeatCoil_2023, "HSPF2 [Btu/W-h]");
         s->pdchDXHeatCoilRegionNum_2023 = newPreDefColumn(state, s->pdstDXHeatCoil_2023, "Region Number");
+        // Std 229 Predef outputs for DX Heating Coils AHRI 2023
+        s->pdchDXHeatCoilMinOADBTforCompOp_2023 =
+            newPreDefColumn(state, s->pdstDXHeatCoil_2023, "Minimum Outdoor Dry-Bulb Temperature for Compressor Operation [C]");
+        s->pdchDXHeatCoilAirloopName_2023 = newPreDefColumn(state, s->pdstDXHeatCoil_2023, "Airloop Name");
```
Comment (Nigusse) [inline src/EnergyPlus/OutputReportPredefined.cc:421]: Adds units to a column heading.

File: src/EnergyPlus/OutputReportPredefined.hh [hunk 8]
```diff
@@ -360,6 +360,8 @@ struct OutputReportPredefinedData : BaseGlobalStruct
     int pdchDXHeatCoilHSPF2SI_2023 = 0;   // HSPF2 value in SI unit at AHRI std. 340/360 conditions [W/W]
     int pdchDXHeatCoilHSPF2IP_2023 = 0;   // HSPF2 value in IP unit at AHRI std. 340/360 conditions [Btu/W-hr]
     int pdchDXHeatCoilRegionNum_2023 = 0; // Region number for which HSPF is calculated
+    int pdchDXHeatCoilMinOADBTforCompOp_2023 = 0;
+    int pdchDXHeatCoilAirloopName_2023 = 0;
```
Comment (Nigusse) [inline src/EnergyPlus/OutputReportPredefined.hh:366]: adds new member variable for the new equipment summary table.

File: src/EnergyPlus/StandardRatings.cc [hunk 9]
```diff
@@ -6885,16 +6888,17 @@ namespace StandardRatings {
...
-                if (state.dataHVACGlobal->StandardRatingsMyHeatOneTimeFlag) {
+                if (state.dataHVACGlobal->StandardRatingsMyHeatOneTimeFlag2) {
                     static constexpr std::string_view Format_992_(
-                        "! <DX Heating Coil Standard Rating Information>, Component Type, Component Name, High Temperature Heating "
+                        "! <DX Heating Coil AHRI 2023 Standard Rating Information>, Component Type, Component Name, High Temperature Heating "
                         "(net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF2 {Btu/W-h}, Region "
                         "Number\n");
                     print(state.files.eio, "{}", Format_992_);
-                    state.dataHVACGlobal->StandardRatingsMyHeatOneTimeFlag = false;
+                    state.dataHVACGlobal->StandardRatingsMyHeatOneTimeFlag2 = false;
```
Comment (Nigusse) [inline src/EnergyPlus/StandardRatings.cc:6898]: This new flag VAR used to create a separate table for HSPF and HSPF2 values

File: tst/EnergyPlus/unit/DXCoils.unit.cc [hunk 10]
```diff
@@ -4656,7 +4656,8 @@ TEST_F(EnergyPlusFixture, TestMultiSpeedCoilsAutoSizingOutput)
  Component Sizing Information, Coil:Heating:DX:MultiSpeed, ASHP HTG COIL, Design Size Resistive Defrost Heater Capacity, 0.00000
 ! <DX Heating Coil Standard Rating Information>, Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF {Btu/W-h}, Region Number
  DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, ASHP HTG COIL, 34415.4, 20666.4, 6.56, 4
- DX Heating Coil Standard Rating Information, Coil:Heating:DX:MultiSpeed, ASHP HTG COIL, 34697.2, 20948.1, 5.12, 4
+! <DX Heating Coil AHRI 2023 Standard Rating Information>, Component Type, Component Name, High Temperature Heating (net) Rating Capacity {W}, Low Temperature Heating (net) Rating Capacity {W}, HSPF2 {Btu/W-h}, Region Number
+ DX Heating Coil AHRI 2023 Standard Rating Information, Coil:Heating:DX:MultiSpeed, ASHP HTG COIL, 34697.2, 20948.1, 5.34, 4
```
Comment (Nigusse) [inline tst/EnergyPlus/unit/DXCoils.unit.cc:4659]: Updated failed unit test.

**Review by Nigusse — COMMENTED** [review]
Code change walkthrough.

File: src/EnergyPlus/DataHVACGlobals.hh [hunk 11]
```diff
@@ -526,6 +526,7 @@ struct HVACGlobalsData : BaseGlobalStruct
     bool StandardRatingsMyCoolOneTimeFlag = true;
     bool StandardRatingsMyCoolOneTimeFlag2 = true;
     bool StandardRatingsMyHeatOneTimeFlag = true;
+    bool StandardRatingsMyHeatOneTimeFlag2 = true;
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataHVACGlobals.hh:535]: This is fine.  Maybe not the greatest of variable names, but not a big problem.

File: src/EnergyPlus/DXCoils.cc (hunk 4 above)
Comment (Myoldmopar) [reply src/EnergyPlus/DXCoils.cc:8701]: Yeah, looks fine. I'll buy it.

File: src/EnergyPlus/OutputReportPredefined.cc (hunk 6 above)
Comment (Myoldmopar) [reply src/EnergyPlus/OutputReportPredefined.cc:412]: I assume this needs an OutputChanges snippet?  Maybe not?  @mjwitte what do you think?

**Review by Myoldmopar — COMMENTED** [review]
Code changes themselves look fine.  My only thought is that we might want to sneak in a description in the OutputChanges file.  Otherwise I am good with this.

**Comment by sebastienbrideau** [comment]
Not that my opinion will have much weight, but I checked this fix, and the results seem to make a lot of sense. I was having issues with HSPF2 and I thought I was missing something so I'm glad it's getting fixed.

The NRC (Canada) has some correlations they gave me that were developed using the AHRI and NEEP databases. 
image
image

I ran a multispeed ASHP for three performances (just varied COP). Before the fix I got this:
image

after the fix:
image

The points without lines (labeled reg) are calculated based on the regression equations above.
It's a much better fit now! Thanks Bereket!

**Comment by Nigusse** [comment]
@Myoldmopar @mjwitte Updated the Output Changes file.

**Comment by Myoldmopar** [comment]
> Not that my opinion will have much weight, but I checked this fix, and the results seem to make a lot of sense. I was having issues with HSPF2 and I thought I was missing something so I'm glad it's getting fixed.

@sebastienbrideau your opinion is quite appreciated!  The more validation we get on any change the better.

@Nigusse thanks for updating the output changes doc.  There's now a tiny conflict in it, so I'll fix that and we should be able to get this merged soon.

**Comment by Myoldmopar** [comment]
@Nigusse the table diffs here are interesting.  When I look locally, here's a sample of what I see in the summary diff:

| Table                                                                 | Big diffs | Small diffs | Equals | String diffs | Size diffs    |
|-----------------------------------------------------------------------|-----------|-------------|--------|--------------|---------------|
| FullName:Equipment Summary_Entire Facility_DX Heating Coils           | 2         | 0           | 7      | 1            | size mismatch |
| FullName:Equipment Summary_Entire Facility_DX Heating Coils [ HSPF2 ] | 0         | 0           | 0      | 0            | not in 2      |
| FullName:Equipment Summary_Entire Facility_DX Heating Coils AHRI 2023 | 0         | 0           | 0      | 0            | not in 1      |

@mjwitte given the issues we just had with EIO/table/flags, would you be able to spot check things here before I move toward a merge?

**Comment by Nigusse** [comment]
The text difference must be related to the following: 

1. Renaming an existing table in equipment summary report,
2. Change column header in an equipment summary table (added units to an existing column header)
3. Added two new columns in an existing equipment summary table

Tabular Value Diffs:
The tabular values change must be related to HSPF and HSPF2 values only. Otherwise, I am not expecting any other Tabular Values change due to code changes in this branch.

**Comment by mjwitte** [comment]
> @mjwitte given the issues we just had with EIO/table/flags, would you be able to spot check things here before I move toward a merge?

@Myoldmopar @Nigusse Looks good here. This can't have the same problem we saw before, because the code for the "DX Heating Coil AHRI 2023 Standard Rating Information" is only in one place (as it should be).

**Comment by Myoldmopar** [comment]
OK, cool, then I'll be merging it shortly.  Thanks!

**Comment by Myoldmopar** [comment]
OK merging this one now.  Thanks @Nigusse and @mjwitte

**Pull request merged**